generator/generator_tool/generator_tool.cpp

```diff
@@ -59,7 +59,10 @@ DEFINE_bool(emit_coasts, false,
 // Generator settings and paths.
 DEFINE_string(osm_file_name, "", "Input osm area file.");
 DEFINE_string(osm_file_type, "xml", "Input osm area file type [xml, o5m].");
-DEFINE_string(data_path, "", "Directory for mwm generation, 'path_to_exe/../../data' if empty.");
+DEFINE_string(data_path, "",
+              "Directory where the generated mwms are put into. Also used as the path for helper "
+              "functions, such as those that calculate statistics and regenerate sections. "
+              "Default: 'path_to_exe/../../data'.");
```

Comment:
Can we use ```getcwd()``` instead of path_to_exe?

Reply:
If I do that I'll add one more place where we stop supporting Windows.

Reply:
Even if we do that. How do you propose to use it inside the macro?

Reply:
Feel free to support it here: https://msdn.microsoft.com/ru-ru/library/sf98bd4y.aspx

Reply:
Yep.

Reply:
Alternative solution is to set
FLAGS_data_path to the WritableDir() just before ParseCommandLineFlags.

Reply:
> Feel free to support it here: https://msdn.microsoft.com/ru-ru/library/sf98bd4y.aspx

Yes, but with one difference. Mac needs `#include <unistd.h>` and Windows needs `#include <direct.h>`.

Reply:
> What's wrong to use a pure function inside macro?

As a variant I can add a function `std::string GetDataPathText() {..}` in anonymous namespace and call GetDataPathText().c_str() instead of the third param of `DEFINE_string`.

Reply:
But anyway, before implementing `std::string GetDataPathText() {..}` and using it we need to think what to do with different includes for getcwd() for Windows and Posix.

Reply:
#ifdef OS_WINDOWS?

Reply:
#ifdef OMIM_OS_WINDOWS

Reply:
Or even better. Why not to include `#include <boost/filesystem.hpp>`?

Reply:
Because if I do so boost *.o files :(

Reply:
Done. PTAL.